lib/project-helpers.ts

```diff
@@ -1,6 +1,50 @@
-export interface CreateProjectValues {
-  name: string;
-  goal: string;
-  owner: string;
-  champion: string;
+import { useEffect } from "react";
+import { useState } from "react";
+import { Project } from "../db/models/project";
+
+export class ProjectValues {
+  public id?: number;
+  public name: string = "";
+  public goal: string = "";
+  public owner: string = "";
+  public champion: string = "";
+  public adminEmails: string[] = [];
+
+  public constructor(init?: Partial<ProjectValues>) {
+    Object.assign(this, init);
+  }
+}
+
+export function useProject(id: string) {
+  const [project, setProject] = useState<ProjectValues>();
+  const [error, setError] = useState<string>();
+  const [isLoading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const getProject = async () => {
+      setLoading(true);
+
+      if (!id) return;
+
+      try {
+        const result = await fetch(`/api/v1/projects/${id}`);
+        const json = await result.json();
```

Comment:
Move this inside the if block below

Reply:
Both the success and error conditions have a JSON payload which is why it's down outside of the `if`. I did go backwards and forwards on this in the stream and I'm not sure the approach I've taken is idea.

Reply:
Makes sense